Makefile

```diff
@@ -109,43 +128,65 @@ GRUB_MKRESCUE = grub-mkrescue
 else
 
 $(BIN): $(DOCKER_STAMP)
-ZIG = $(DOCKER_CMD) run --rm -w /build -v $(NAME):/build:rw -ti zig zig
+ZIG = $(DOCKER_CMD) run --rm -w /build -v $(NAME):/build:rw zig zig
 
 $(ISO): $(DOCKER_STAMP)
-GRUB_MKRESCUE = $(DOCKER_CMD) run --rm -w /build -v $(NAME):/build:rw -ti zig grub-mkrescue
+GRUB_MKRESCUE = $(DOCKER_CMD) run --rm -w /build -v $(NAME):/build:rw zig grub-mkrescue
+
+format: $(DOCKER_STAMP)
 
 endif
 
-$(ISO): $(BIN) $(GRUB_CONF)
+$(ISO): $(BIN) $(GRUB_CONF) | $(CI_STAMP).$(CI)
 	$(GRUB_MKRESCUE) --compress=xz -o $@ $(ISODIR)
 
 run: $(ISO)
-	qemu-system-$(ARCH) -cdrom $<
+	qemu-system-$(ARCH) -cdrom $< $(QEMU_ARGS)
 
 run_kernel: $(BIN)
 	qemu-system-$(ARCH) -kernel $<
 
-$(DOCKER_STAMP): dockerfile
-	$(DOCKER_CMD) build -t zig .
-	$(DOCKER_CMD) volume create --name $(NAME) --driver=local --opt type=none --opt device=$(PWD) --opt o=bind,uid=$(shell id -u)
+$(DOCKER_STAMP): dockerfile | docker_volume
+	$(DOCKER_CMD) build -t zig . -f dockerfile
 	> $(DOCKER_STAMP)
 
+docker_volume:
+	$(DOCKER_CMD) volume rm $(NAME) || true
+	$(DOCKER_CMD) volume create --name $(NAME) --driver=local --opt type=none --opt device=$(PWD) --opt o=bind,uid=$(shell id -u)
+
+$(CI_STAMP).$(CI):
+	rm -rf $(CI_STAMP).*
+	touch $@
```

Comment:
This is a simple rule for making iso depend on the CI variable.
I didn't really find an efficient way to make a rule depend on a variable :eyes:

Moreover, it's the bin rule that should depend on it, not the iso one (since it already depends on the bin one).

Reply:
Obsolete since the build logic was widely reworked with #140